app/models/concerns/due_date_permissions.rb

```diff
@@ -0,0 +1,199 @@
+# frozen_string_literal: true
+
+module DueDatePermissions
+  extend ActiveSupport::Concern
+
+  # Permission checking methods that combine deadline-based and role-based logic
+  # These methods provide a unified interface for checking if actions are allowed
+
+  def can_submit?
+    return false unless submission_allowed_id
+
+    deadline_right = DeadlineRight.find_by(id: submission_allowed_id)
+    deadline_right&.name&.in?(%w[OK Late])
+  end
+
+  def can_review?
+    return false unless review_allowed_id
+
+    deadline_right = DeadlineRight.find_by(id: review_allowed_id)
+    deadline_right&.name&.in?(%w[OK Late])
+  end
+
+  def can_take_quiz?
+    return false unless quiz_allowed_id
+
+    deadline_right = DeadlineRight.find_by(id: quiz_allowed_id)
+    deadline_right&.name&.in?(%w[OK Late])
+  end
+
+  def can_teammate_review?
+    return false unless teammate_review_allowed_id
+
+    deadline_right = DeadlineRight.find_by(id: teammate_review_allowed_id)
+    deadline_right&.name&.in?(%w[OK Late])
+  end
+
+  def can_metareview?
+    return false unless respond_to?(:metareview_allowed_id) && metareview_allowed_id
+
+    deadline_right = DeadlineRight.find_by(id: metareview_allowed_id)
+    deadline_right&.name&.in?(%w[OK Late])
+  end
+
+  # Generic permission checker that can be extended for any action
```

Comment:
This is already defined in DueDateActions.  Really confused about why it's also in this class.